go/fn/doc.go

```diff
@@ -13,58 +13,66 @@
-ResourceList.
+Before you start
 
-The ResourceList type and the KubeObject type are the core part of this package.
+This fn SDK requires some basic KRM function Specification knowledge. To make the best usage of your time, we recommend
+you to be familiar with "ResourceList" before moving forward.
+
+  The KRM Function Specification, or "ResourceList", defines the standards of the inter-process communication between
+  the orchestrator (i.e. kpt CLI) and functions.
+
+It is created by kpt project and later on donated to CNCF.
+See KRM Function Specification reference in https://github.com/kubernetes-sigs/kustomize/blob/master/cmd/config/docs/api-conventions/functions-spec.md
+
+KRM Function
+
+A KRM function can mutate and/or validate Kubernetes resources in a ResourceList.
+
+The ResourceList type and the KubeObject type are the core parts of this package.
 The ResourceList type maps to the ResourceList in the function spec.
 
-The KubeObject represent a kubernetes resource in a ResourceList, and it's the basic
-unit to perform most CRUD operations.
-
-A KRM function does the following things:
-  1. read yaml bytes from stdin and convert it to a ResourceList
-  2. perform mutation and validation on the resources in the ResourceList
-  3. write the updated ResourceList out to stdout in yaml format
-  4. Any diagnostic messages should be written to stderr
-
-In most cases, you only need to do #2 which is to use pre-defined ResourceListProcessor or use your own ResourceListProcessor
- and then pass it to AsMain. In the following example, we
-use a struct which implements the fn.Runner `Run` method
-
-```go
-package main
-
-import (
-	"fmt"
-	"github.com/GoogleContainerTools/kpt-functions-sdk/go/fn"
-)
-
-var _ fn.Runner = &YourKRMFn{}
-
-type YourKRMFn struct {
-	Field1 map[string]string `json:",inline,omitempty"`
-}
-
-// Add your function logic in run.
-// `resourcelist.functionConfig` is assigned to YourKRMFn object. Your functionConfig kind shall be `YourKRMFn` or `ConfigMap` (limited usage)
-// `resourcelist.items` is the passed-in `items` parameter.
-// `resourcellist.result` is empty, you can add result via `ctx` methods like `AddGeneralResult`, `AddErrResultAndDie`.
-func (r *YourKRMFn) Run(ctx *fn.Context, functionConfig *fn.KubeObject, items []*fn.KubeObject) {
-	for _, o := range items {
-		// Your code
-	}
-}
-
-func main() {
-	if err := fn.AsMain(&YourKRMFn{}); err != nil {
-		os.Exit(1)
-	}
-}
-```
+Read more about how to use KRM functions in https://kpt.dev/book/04-using-functions/
+
+Read more about how to develop a KRM function in https://kpt.dev/book/05-developing-functions/
+
+A general workflow is:
+  1. Reads the "ResourceList" object from STDIN.
+  2. Gets the function configs from the "ResourceList.FunctionConfig".
+  3. Mutate or validate the Kubernetes YAML resources from the "ResourceList.Items" field with the function configs.
+  4. Writes the modified "ResourceList" to STDOUT.
```

Comment:
Maybe add an step about writing `Results` in resourceList

Reply:
Done